src/GazeboThruster.cpp

```diff
@@ -140,14 +146,17 @@ void GazeboThruster::readInput(ThrustersMSG const& thrustersMSG)
     for(int i = 0; i < thrustersMSG->thrusters_size(); ++i)
     {
         bool thrusterFound = false;
+        double disturbance = 0.0;
         const gazebo_thruster::msgs::Thruster& thrusterCMD = thrustersMSG->thrusters(i);
         for(vector<Thruster>::iterator thruster = thrusters.begin();
                 thruster != thrusters.end(); ++thruster)
         {
             if(thrusterCMD.name() == thruster->name)
             {
                 thrusterFound = true;
-                thruster->effort = updateEffort(thrusterCMD);
+                disturbance = sampleNormal(thruster->disturbance.mu, 
+                    thruster->disturbance.sigma);
+                thruster->effort = updateEffort(thrusterCMD, disturbance);
```

Comment:
Is this really what you want ?

I would have applied the disturbance at the simulation tick, not at the input.

Reply:
> Hey. This is a common package, not at all owned (or private to) 13.
> 
> I think that what you want to do is make your private pull request depend on this one, not the other way around (this does not depend on the bundle, does it really ?)

Sorry, @doudou. This was my bad, I have accidentally chosen the wrong target to perform the merge, it was intended to be done on 13's fork of this common package. And you are right, the dependency works the other way around, thanks.

Reply:
> Is this really what you want ?
> 
> I would have applied the disturbance at the simulation tick, not at the input.

The idea is really to add a disturbance on the thruster effort that will be transmitted to the robot so that the controllers robustness could be assessed. This kind of disturbance could physically represent unexpected behaviors on the mechanic assembly for example. I am not sure that I understood how to apply the disturbance on the simulation tick.

Reply:
> I would have applied the disturbance at the simulation tick, not at the input.
I think he meant that he would put it at the "updateHook" at the orogen Task, or at the "updateBegin" at the library.

Reply:
> > Hey. This is a common package, not at all owned (or private to) 13.
> > I think that what you want to do is make your private pull request depend on this one, not the other way around (this does not depend on the bundle, does it really ?)
> 
> Sorry, @doudou. This was my bad, I have accidentally chosen the wrong target to perform the merge, it was intended to be done on 13's fork of this common package. And you are right, the dependency works the other way around, thanks.

Thanks, Sylvain. Gabriel, you can click Edit and fix the PR to target the fork.

Reply:
> > Hey. This is a common package, not at all owned (or private to) 13.
> > I think that what you want to do is make your private pull request depend on this one, not the other way around (this does not depend on the bundle, does it really ?)
> 
> Sorry, @doudou. This was my bad, I have accidentally chosen the wrong target to perform the merge, it was intended to be done on 13's fork of this common package. And you are right, the dependency works the other way around, thanks.

Suit yourself. But I like the idea of integrating that in the common package ... Your choice.

Reply:
Cool, let's integrate that on the common package.